File: platforms/flutter/ios/Classes/MermanFlutterPlugin.swift
import Flutter
import UIKit

/// Minimal Flutter plugin registration for merman.
///
/// Rendering is handled by Dart FFI. The static merman-ffi library is linked
/// through the vendored XCFramework, so no method channels are needed.
public class MermanFlutterPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        // FFI-only plugin: no channels to register.
    }
}

File: platforms/flutter/windows/merman_flutter_plugin.cpp
// Flutter plugin registration for Merman on Windows.
//
// All rendering is done via Dart FFI. The plugin DLL is a registration stub;
// merman_ffi.dll is bundled beside it by CMake.

#include "merman/merman_flutter_plugin.h"

#include <flutter/plugin_registrar_windows.h>

namespace {

class MermanFlutterPlugin : public flutter::Plugin {
 public:
  static void RegisterWithRegistrar(flutter::PluginRegistrarWindows* registrar) {
    // FFI-only plugin: no channels to register.
  }

  MermanFlutterPlugin() = default;
  virtual ~MermanFlutterPlugin() = default;
  MermanFlutterPlugin(const MermanFlutterPlugin&) = delete;
  MermanFlutterPlugin& operator=(const MermanFlutterPlugin&) = delete;
};

}  // namespace

void MermanFlutterPluginRegisterWithRegistrar(
    FlutterDesktopPluginRegistrarRef registrar) {
  MermanFlutterPlugin::RegisterWithRegistrar(
      flutter::PluginRegistrarManager::GetInstance()
          ->GetRegistrar<flutter::PluginRegistrarWindows>(registrar));
}
